src/handler.rs

```diff
@@ -148,7 +155,32 @@ pub enum Command {
 
 #[derive(Debug, Clone, Deserialize)]
 #[serde(rename_all = "camelCase")]
-pub struct SubstrateCommand {}
+pub enum SubstrateCommand {
+    MixerRelayTx(MixerRelayTransaction),
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MixerRelayTransaction {
+    /// one of the supported chains of this relayer
+    pub chain: String,
+    /// The tree id of the mixer's underlying tree
+    pub id: u32,
+    /// The zero-knowledge proof bytes
+    pub proof: Vec<u8>,
+    /// The target merkle root for the proof
+    pub root: Element,
+    /// The nullifier_hash for the proof
+    pub nullifier_hash: Element,
+    /// The receipient of the transaction
+    pub recipient: AccountData,
+    /// The relayer of the transaction
+    pub relayer: AccountData,
+    /// The relayer's fee for the transaction
+    pub fee: u128,
```

Comment:
prefer `U256` over the normal `u128` as it can take bigger numbers plus it can be sent as hex, making life easier when working from the front-end.
```suggestion
    pub fee: U256,
```

Reply:
But the balance on chain is a u128?

Reply:
I guess you have to think about how you do send it to the withdraw, if it is u128 then yes, it is all good, if it is a 32bytes array, then we need to change it to that.